website/docs/r/mssql_server_transparent_data_encryption.html.markdown

```diff
@@ -0,0 +1,178 @@
+    storage_account_access_key              = azurerm_storage_account.example.primary_access_key
+    storage_account_access_key_is_secondary = true
+    retention_in_days                       = 6
+  }
+
+  tags = {
+    environment = "production"
+  }
+
+  identity {
+    type = "SystemAssigned"
+  }
+}
+
+# Create a key vault with policies for the deployer to create a key & SQL Server to wrap/unwrap/get key
+resource "azurerm_key_vault" "example" {
+  name                        = "example"
+  location                    = azurerm_resource_group.example.location
+  resource_group_name         = azurerm_resource_group.example.name
+  enabled_for_disk_encryption = true
+  tenant_id                   = data.azurerm_client_config.current.tenant_id
+  soft_delete_retention_days  = 7
+  purge_protection_enabled    = false
+
+  sku_name = "standard"
+
+  access_policy {
+    tenant_id = data.azurerm_client_config.current.tenant_id
+    object_id = data.azurerm_client_config.current.object_id
+
+    key_permissions = [
+      "Get", "List", "Create", "Delete", "Update", "Recover", "Purge",
+    ]
+  }
+  access_policy {
+    tenant_id = azurerm_mssql_server.example.identity[0].tenant_id
+    object_id = azurerm_mssql_server.example.identity[0].principal_id
+
+    key_permissions = [
+      "Get", "WrapKey", "UnwrapKey"
+    ]
+  }
+}
+
+resource "azurerm_key_vault_key" "example" {
+  name         = "byok"
+  key_vault_id = azurerm_key_vault.example.id
+  key_type     = "RSA"
+  key_size     = 2048
+
+  key_opts = [
+    "unwrapKey",
+    "wrapKey",
+  ]
+
+  depends_on = [
+    azurerm_key_vault.example
+  ]
+}
+
+resource "azurerm_mssql_server_transparent_data_encryption" "example" {
+  server_id        = azurerm_mssql_server.example.id
+  key_vault_key_id = azurerm_key_vault_key.example.id
+}
+
+
+```
+
+## Arguments Reference
+
+The following arguments are supported:
+
+* `server_id` - (Required) Specifies the name of the MS SQL Server.
+
+---
+
+* `key_vault_key_id` - (Optional) To use customer managed keys from Azure Key Vault, provide the AKV Key ID. To use service managed keys, omit this field.
+
+~> **NOTE:** In order to use customer managed keys, the identity of the MSSQL server must have the following permissions on the key vault: 'get', 'wrapKey' and 'unwrapKey' 
+
+## Attributes Reference
+
+In addition to the Arguments listed above - the following Attributes are exported: 
+
+* `id` - The ID of the MSSQL encryption protector
+
+## Timeouts
+
+The `timeouts` block allows you to specify [timeouts](https://www.terraform.io/docs/configuration/resources.html#timeouts) for certain actions:
+
+* `create` - (Defaults to 30 minutes) Used when creating the MSSQL.
+* `read` - (Defaults to 5 minutes) Used when retrieving the MSSQL.
+* `update` - (Defaults to 30 minutes) Used when updating the MSSQL.
+* `delete` - (Defaults to 30 minutes) Used when deleting the MSSQL.
+
```

Comment:
Can we add an Import section to the end here?

````suggestion

## Import

SQL Server Transparent Data Encryption can be imported using the resource id, e.g.

```shell
terraform import azurerm_mssql_server_transparent_data_encryption.example /subscriptions/12345678-1234-9876-4563-123456789012/resourceGroups/group1/providers/Microsoft.Sql/servers/server1/encryptionProtector/current
```

````

Reply:
I'm curious about this one. Encryption Protector is automatically created as part of SQL Server. The intent was to disable importing and instead allow this resource to overwrite the existing EP. You'll notice the typical ImportAsExists error has been removed. If we force importing the resource, it means you won't be able to create SQL Server & create the EP in the same configuration since that error will be thrown as a result of MS SQL server creating it under the covers. To support importing, you'd have to:

1. Create SQL Server
2. Import Transparent Data Encryption resource
3. Update TDE (Encryption Protector) to your desired settings

To me it made more sense to disable importing of this resource. If so, should the Importer property on line 30 `Importer: azSchema.ValidateResourceIDPriorToImport(func(id string) error` be set to nil instead?

Reply:
You're absolutely right, I missed that detail when considering the import of an already managed SQL Server instance - and that's not a viable workflow that we'd want to inflict. Perhaps we should add a short note in that section stating "Import is not required for this resource"?

I think we can leave the existing Importer func in place, as there's no reason _not_ to allow import if someone tries to do that?